db/controllers/news.controller.js

```diff
@@ -57,10 +58,32 @@ function getCommentsById(req, res, next) {
     });
 }
 
+function postCommentById(req, res, next) {
+  const newComment = req.body;
+  if (
+    newComment.username === undefined ||
+    newComment.body === undefined ||
+    typeof newComment.username !== "string" ||
+    typeof newComment.body !== "string"
+  ) {
+    res.status(400).send({ error: "Bad Request" });
+  }
+  newComment.article_id = req.params.article_id;
+
+  insertCommentById(newComment)
+    .then(() => {
+      res.status(201).send({ comment: newComment });
+    })
+    .catch((err) => {
+      next(err);
+    });
+}
```

Comment:
Your database will throw an error if something goes wrong, and it will be caught in your .catch block. There is no need for checking the username and body here.